clients/roscpp/src/libros/init.cpp

```diff
@@ -447,12 +452,6 @@ void check_ipv6_environment() {
 
 void init(const M_string& remappings, const std::string& name, uint32_t options)
 {
-  if (!g_atexit_registered)
-  {
-    g_atexit_registered = true;
-    atexit(atexitCallback);
```

Comment:
I would expect you to call `initShutdownCaller()` here too. 
`shutdownCallback()` is only called by an explicit remote shutdown request, isn't it?

Reply:
replaced with the `installDeinitCaller` call at the end of the function